source/frontend/src/features/mapSideBar/lease/tabs/consultations/detail/ConsultationListView.tsx

```diff
@@ -0,0 +1,222 @@
+import { useMemo } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaPlus, FaTrash } from 'react-icons/fa';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { StyledRemoveLinkButton } from '@/components/common/buttons/RemoveButton';
+import ContactFieldContainer from '@/components/common/ContactFieldContainer';
+import EditButton from '@/components/common/EditButton';
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { StyledSummarySection } from '@/components/common/Section/SectionStyles';
+import { SectionListHeader } from '@/components/common/SectionListHeader';
+import * as API from '@/constants/API';
+import Claims from '@/constants/claims';
+import useKeycloakWrapper from '@/hooks/useKeycloakWrapper';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { getDeleteModalProps, useModalContext } from '@/hooks/useModalContext';
+import { ApiGen_Concepts_ConsultationLease } from '@/models/api/generated/ApiGen_Concepts_ConsultationLease';
+import { prettyFormatDate } from '@/utils';
+import { booleanToYesNoUnknownString } from '@/utils/formUtils';
+
+export interface IConsultationListViewProps {
+  loading: boolean;
+  consultations: ApiGen_Concepts_ConsultationLease[];
+  onAdd: () => void;
+  onEdit: (consultationId: number) => void;
+  onDelete: (consultationId: number) => void;
+}
+
+interface GroupedConsultations {
+  consultationTypeCode: string;
+  consultationTypeDescription: string;
+  consultations: ApiGen_Concepts_ConsultationLease[];
+  hasItems: boolean;
+}
+
+export const ConsultationListView: React.FunctionComponent<IConsultationListViewProps> = ({
+  loading,
+  consultations,
+  onAdd,
+  onEdit,
+  onDelete,
+}) => {
+  const keycloak = useKeycloakWrapper();
+  const history = useHistory();
+  const match = useRouteMatch();
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const { getOptionsByType } = useLookupCodeHelpers();
+  const consultationTypeCodes = getOptionsByType(API.CONSULTATION_TYPES);
+
+  const groupedConsultations = useMemo(() => {
+    const grouped: GroupedConsultations[] = [];
+    consultationTypeCodes.forEach(ct =>
+      grouped.push({
+        consultationTypeCode: ct.value.toString(),
+        consultationTypeDescription: ct.label,
+        consultations: consultations.filter(c => c.consultationTypeCode.id === ct.value),
+        hasItems: consultations.filter(c => c.consultationTypeCode.id === ct.value).length === 0,
+      }),
+    );
+    return grouped;
+  }, [consultationTypeCodes, consultations]);
+
+  if (loading) {
+    return <LoadingBackdrop show={loading} parentScreen={true} />;
```

Comment:
You probably couldn't see this, because load times are so fast on our locals, but this is what the loading screen looks like:

![image](https://github.com/user-attachments/assets/e40a470c-db53-4058-b310-0d331a19b927)

My preference for these is to render an empty version of the screen with the loading backdrop overtop, but some kind of correction required here.

Reply:
I think there is some room for improvement here. Will chat win ana to check what her thoughts are on loading.